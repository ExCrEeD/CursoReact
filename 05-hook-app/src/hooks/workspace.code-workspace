{
	"folders": [
		{
			"path": "..\\..\\..\\07-heroes-app"
		}
	],
	"settings": {}
}