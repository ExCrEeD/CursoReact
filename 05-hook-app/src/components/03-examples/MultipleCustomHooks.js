import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch'
import "../02-useEffect/effects.css"
export const MultipleCustomHooks = () => {

    const {counter,increment} = useCounter(1);
    const {loading,data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    const {author,quote} = !!data && data[0];
    return (
        <div>
            <h1>BreakingBar Quotes!!</h1>
            <hr/>
            {
                loading?( 
                <div className="alert alert-info text-center">
                    loading..
                </div>
                ):(
                <blockquote className="blockquote text-right">
                    <p className="mb-0">{quote}</p>
                    <br/>
                    <footer className="blockquote-footer">{author}</footer>
                </blockquote>
            )
            }
            <button className="btn-primary" onClick={increment}>siguiente quote</button>
           
            
        </div>
    )
}
