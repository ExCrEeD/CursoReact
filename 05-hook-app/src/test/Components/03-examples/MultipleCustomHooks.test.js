import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import { useFetch } from "../../../hooks/useFetch";

jest.mock("../../../hooks/useFetch");

describe('pruebas en multipleCustomHooks', () => {
    test('debe de mostarse correctamente ', () => {
        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error:null
        });
        const wrapper = shallow(<MultipleCustomHooks/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('should debe de mostar la informacion',()=>{
        useFetch.mockReturnValue({
            data:[{
                author:'daniel',
                quote:'Hola mundo'
            }],
            loading:false,
            error:null
        });
        const wrapper = shallow(<MultipleCustomHooks/>);
        console.log(wrapper.html());
        const loadingExist = wrapper.find('.alert').exists();
        expect(loadingExist).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo');
        expect(wrapper.find('footer').text().trim()).toBe('daniel');
    })
    
    
})
