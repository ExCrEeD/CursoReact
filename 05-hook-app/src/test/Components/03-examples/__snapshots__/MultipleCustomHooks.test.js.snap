// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en multipleCustomHooks debe de mostarse correctamente  1`] = `
<div>
  <h1>
    BreakingBar Quotes!!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading..
  </div>
  <button
    className="btn-primary"
    onClick={[Function]}
  >
    siguiente quote
  </button>
</div>
`;
